src/beast/beast/asio/bind_handler.h

```diff
@@ -23,142 +23,119 @@
 #include <boost/asio/detail/handler_alloc_helpers.hpp>
 #include <boost/asio/detail/handler_cont_helpers.hpp>
 #include <boost/asio/detail/handler_invoke_helpers.hpp>
-
 #include <functional>
 #include <type_traits>
 #include <utility>
 
 namespace beast {
-namespace asio {
 
 namespace detail {
 
 /** Nullary handler that calls Handler with bound arguments.
-    The rebound handler provides the same io_service execution
+
+    The bound handler provides the same io_service execution
     guarantees as the original handler.
 */
-template <class DeducedHandler, class... Args>
+template<class DeducedHandler, class... Args>
 class bound_handler
 {
 private:
     using args_type = std::tuple <std::decay_t <Args>...>;
+    
+    std::decay_t<DeducedHandler> h_;
+    args_type args_;
 
-    std::decay_t <DeducedHandler> m_handler;
-    args_type m_args;
-
-    template <class Handler, class Tuple, std::size_t... S>
-    static void invoke (Handler& h, Tuple& args,
+    template<class Handler, class Tuple, std::size_t... S>
+    static void invoke(Handler& h, Tuple& args,
         std::index_sequence <S...>)
     {
-        h (std::get <S> (args)...);
+        h(std::get<S>(args)...);
     }
 
 public:
     using result_type = void;
 
     explicit
-    bound_handler (DeducedHandler&& handler, Args&&... args)
-        : m_handler (std::forward <DeducedHandler> (handler))
-        , m_args (std::forward <Args> (args)...)
-    {
-    }
-
-    void
-    operator() ()
-    {
-        invoke (m_handler, m_args,
-            std::index_sequence_for <Args...> ());
-    }
-
-    void
-    operator() () const
+    bound_handler(DeducedHandler&& handler, Args&&... args)
+        : h_(std::forward<DeducedHandler>(handler))
```

Comment:
`DeducedHandler` is not an inferred type here, so it is not a forwarding ref. Both `handler` and `args` will be rvalue refs, so `std::move` instead of `std::forward`.